content/en/docs/concepts/services-networking/_index.md

```diff
@@ -7,58 +7,102 @@ description: >
-- Cluster networking provides communication between different Pods.
-- The [Service](/docs/concepts/services-networking/service/) API lets you
-  [expose an application running in Pods](/docs/tutorials/services/connect-applications-service/)
-  to be reachable from outside your cluster.
-  - [Ingress](/docs/concepts/services-networking/ingress/) provides extra functionality
-    specifically for exposing HTTP applications, websites and APIs.
-  - [Gateway API](/docs/concepts/services-networking/gateway/) is an {{<glossary_tooltip text="add-on" term_id="addons">}}
-    that provides an expressive, extensible, and role-oriented family of API kinds for modeling service networking.
-- You can also use Services to
-  [publish services only for consumption inside your cluster](/docs/concepts/services-networking/service-traffic-policy/).
+The Kubernetes network model is built out of several pieces:
+
+  * Each {{< glossary_tooltip text="pod" term_id="pod" >}} in a cluster gets its
+    own unique cluster-wide IP address.
+
+      * A pod has its own private network namespace which is shared by
+        all of the containers within the pod. Processes running in
+        different containers in the same pod can communicate with each
+        other over `localhost`.
+
+  * The _pod network_ (also called a cluster network) handles communication
+    between pods. It ensures that (barring intentional network
+    segmentation):
+
+      * All pods can communicate with all other pods, whether they are
+        on the same [node](/docs/concepts/architecture/nodes/) or on
+        different nodes. Pods can communicate with each other
+        directly, without the use of proxies or address translation
+        (NAT).
+
+          * (On Windows, this rule does not apply to host-network pods.)
+
+      * Agents on a node (such as system daemons, or kubelet) can
+        communicate with all pods on that node.
+
+  * The [Service](/docs/concepts/services-networking/service/) API
+    lets you provide a stable (long lived) IP address or hostname for a service implemented
+    by one or more backend pods, where the individual pods making up
+    the service can change over time.
+
+      * Kubernetes automatically manages
+        [EndpointSlice](/docs/concepts/services-networking/endpoint-slices/)
+        objects to provide information about the pods currently
+        backing a Service.
+
+      * A _service proxy_ implementation takes care of converting information from
+        Service and EndpointSlice objects into packet-rewriting
+        rules using appropriate operating system APIs.
+
+  * The [Gateway](/docs/concepts/services-networking/gateway/) API
+    (or its predecessor,
+    [Ingress](/docs/concepts/services-networking/ingress/)) allows
+    you to make Services accessible to clients that are outside the cluster.
+
+      * A simpler, but less-configurable, mechanism for cluster
```

Comment:
Are we comfortable implying Service LB is obsolete?  Does L4 Gateway handle EVERYTHING that LB does?  I don't think so...

Reply:
What change would you recommend?

I definitely wasn't trying to imply that it's obsolete (and note that the old version of this page doesn't link to the Service LB docs at all, so this is actually making that feature _more_ prominent).

But if we're going to have a single top-level answer for "how cluster ingress works in Kubernetes", it should be Gateway, right?

(I could demote the Ingress reference to a nested bullet point... or remove it.)

Reply:
>  if we're going to have a single top-level answer for "how cluster ingress works in Kubernetes", it should be Gateway, right?

I can't see how that is true until L4 gateway is as complete as service LB?  @robscott 

As for wording here, something like:

"An alternative mechanism for exposing a single Service is the Service API's type: LoadBalancer.  It is simpler than Gateway, but less powerful."

Or something?  I know we don't want to make this "at the time of this writing" but the reality is that, at the time of this writing Gateway does not fully replace type: LB.

Reply:
I think this PR makes a better compromise than the page as it stands (and obviously so), so I strongly support merging as-is and then iterating.

I don't see anything that is so wrong that it would seriously mislead readers or get in the way of their learning.